allennlp/fairness/fairness_metrics.py

```diff
@@ -0,0 +1,466 @@
+from typing import Optional, Dict, Union
+
+import torch
+from torch.distributions.categorical import Categorical
+from torch.distributions.kl import kl_divergence
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.training.metrics.metric import Metric
+
+"""
+Fairness metrics are based on:
+1) Barocas, S.; Hardt, M.; and Narayanan, A. 2019. Fairness and machine learning. fairmlbook.org.
+2) Zhang, B. H.; Lemoine, B.; and Mitchell, M. 2018. Mitigating unwanted biases with adversarial learning.
+In Proceedings of the 2018 AAAI/ACM Conference on AI, Ethics, and Society, 335-340.
+3) Hardt, M.; Price, E.; Srebro, N.; et al. 2016. Equality of opportunity in supervised learning.
+In Advances in Neural Information Processing Systems, 3315–3323.
+4) Beutel, A.; Chen, J.; Zhao, Z.; and Chi, E. H. 2017. Data decisions and theoretical implications when
+adversarially learning fair representations. arXiv preprint arXiv:1707.00075.
+5) Aka, O.; Burke, K.; Bäuerle, A.; Greer, C.; and Mitchell, M. 2021.
+Measuring model biases in the absence of ground truth. arXiv preprint arXiv:2103.03417.
+
+It is provably impossible to satisfy any two of Independence, Separation, and Sufficiency simultaneously,
+except in degenerate cases.
+"""
+
+
+@Metric.register("independence")
+class Independence(Metric):
+    """
+    Independence. Assumes integer labels, with
```

Comment:
It might be helpful to add a small example of how the output can be interpreted in a real-world context. Eg. kl div of nearly 0 for some protected variable implies independence, which is useful in a certain context.